server/sonar-web/src/main/js/apps/marketplace/App.tsx

```diff
@@ -0,0 +1,252 @@
+  mounted: boolean;
+  timer?: NodeJS.Timer;
+
+  static contextTypes = {
+    router: PropTypes.object.isRequired
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      loadingEditions: true,
+      loadingPlugins: true,
+      pending: {
+        installing: [],
+        updating: [],
+        removing: []
+      },
+      plugins: []
+    };
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+    this.fetchEditions();
+    this.fetchPendingPlugins();
+    this.fetchEditionStatus();
+    this.fetchQueryPlugins();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.location.query.filter !== this.props.location.query.filter) {
+      this.fetchQueryPlugins();
+    }
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  fetchQueryPlugins = () => {
+    const query = parseQuery(this.props.location.query);
+    if (query.filter === 'updates') {
+      this.fetchUpdatesOnly();
+    } else {
+      this.fetchAllPlugins();
+    }
+  };
+
+  fetchAllPlugins = () => {
+    this.setState({ loadingPlugins: true });
+    Promise.all([getInstalledPluginsWithUpdates(), getAvailablePlugins()]).then(
+      ([installed, available]) => {
+        if (this.mounted) {
+          this.setState({
+            loadingPlugins: false,
+            plugins: sortBy(uniqBy([...installed, ...available.plugins], 'key'), 'name')
+          });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loadingPlugins: false });
+        }
+      }
+    );
+  };
+
+  fetchUpdatesOnly = () => {
+    this.setState({ loadingPlugins: true });
+    getPluginUpdates().then(
+      plugins => {
+        if (this.mounted) {
+          this.setState({ loadingPlugins: false, plugins });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loadingPlugins: false });
+        }
+      }
+    );
+  };
+
+  fetchPendingPlugins = () =>
+    getPendingPlugins().then(
+      pending => {
+        if (this.mounted) {
+          this.setState({ pending });
+        }
+      },
+      () => {}
+    );
+
+  fetchEditionStatus = () =>
+    getEditionStatus().then(
+      editionStatus => {
+        if (this.mounted) {
+          this.updateEditionStatus(editionStatus);
+        }
+      },
+      () => {}
+    );
+
+  fetchEditions = () => {
+    this.setState({ loadingEditions: true });
+    getEditionsList(this.props.editionsUrl).then(
+      editionsPerVersion => {
+        if (this.mounted) {
+          this.setState({
+            editions: getEditionsForVersion(editionsPerVersion, this.props.sonarqubeVersion),
+            loadingEditions: false
+          });
+        }
+      },
+      () => {
+        if (this.mounted) {
+          this.setState({ loadingEditions: false });
+        }
+      }
+    );
+  };
+
+  updateEditionStatus = (editionStatus: EditionStatus) => {
+    this.setState({ editionStatus: editionStatus });
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected property shorthand in object literal ('{editionStatus}'). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3498)